libs/@blockprotocol/graph/src/stdlib/bound.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  NonNullTimeInterval,
+  TemporalBound,
+} from "../types/temporal-versioning.js";
+
+export const compareBounds = (
+  lhs: TemporalBound,
+  rhs: TemporalBound,
+  lhsType: keyof NonNullTimeInterval,
+  rhsType: keyof NonNullTimeInterval,
+): number => {
+  // If the bound values are not equal, then the bound with the lower value is less than the bound with the higher
+  // value.
+  if (lhs.kind !== "unbounded" && rhs.kind !== "unbounded") {
+    if (lhs.limit !== rhs.limit) {
+      return lhs.limit.localeCompare(rhs.limit);
+    }
+  }
```

Comment:
As you don't use a `match` pattern, this function can be written more efficiently: After checking for `"unbounded"` in line 14, you can do any `"unbounded"` comparison before invoking any comparison operator at all. This means,
- if both bounds are `"unbounded"`
    - and both `.kind`s are equal:  the bounds are equal.
    - if `lhs.kind` is `"start"`: `lhs` is smaller
    - else `rhs` is smaller
- if `lhs` is `"unbounded"`
    - if `lhsType` is `"start"`: `lhs` is smaller
    - else `rhs` is smaller
- if `rhs` is `"unbounded"`
    - if `rhsType` is `"start"`: `rhs` is smaller
    - else `lhs` is smaller

With that, all possibilities with `"unbounded"` are captured. Then you can continue with the logic when the limits are equal. 
I don't ask to change this here as the corresponding Rust code is very similar but very well tested and writing the same tests again is a little bit too much. Ideally, I'd favor to have this functionality exported from a Rust crate (this applies to all interval-related operations).

Reply:
I'm going to opt to not change it right now as from reading it, I'm not sure if your suggestion will end up more efficient, or clearer to read. And as you highlight, we'd like to be certain about correctness here first.

Reply:
The approach is more efficient, but only by a very small margin, and in our use case, this will very very unlikely have any impact, so I agree to keep the logic as it is.